feat: allow to define the ports to open when podifying (#1017)

lstocchi opened the pull request:
### What does this PR do?

it allow users to define the ports to open when podifying 

### Screenshot/screencast of this PR

![expose-port-pod](https://user-images.githubusercontent.com/49404737/234527674-ed14f52c-8f66-445f-8038-c8135bb61136.gif)

### What issues does this PR fix or reference?

it fixes #1017 

### How to test this PR?

1. select 1+ containers with ports
2. deselect all ports or some of them
3. create a pod
4. verify that the ports you deselected are not opened

What should it happen when the user selects two containers with the same port? Only one container is displayed in the list and an error message explains what happened?? Only show the warning??

@mairin as i added a new part I also updated the rest with your design https://github.com/containers/podman-desktop/issues/2183 . As the port list is a new stuff, let me know if that's ok or i need to change it

benoitf commented on packages/renderer/src/lib/pod/PodCreateFromContainers.svelte:
```diff
@@ -12,7 +12,7 @@ import ContainerIcon from '../images/ContainerIcon.svelte';
 let podCreation: PodCreation;
 let createInProgress = false;
-let createError = undefined;
+$: createError = undefined;
```
It's not clear to me why it's in reactive form ?

lstocchi replied:
old stuff, reverted

jeffmaury commented on packages/renderer/src/lib/container/ContainerDetailsSummary.svelte:
```diff
@@ -3,6 +3,16 @@ import type { ContainerInfoUI } from './ContainerInfoUI';
 
 export let container: ContainerInfoUI;
+
+function printPorts(ports: number[]): string {
```
This kind of function should be in a utility and not in Svelte/UI

jeffmaury reviewed (approved):
LGTM

slemeur commented:
👍

mairin commented:
@lstocchi The port list looks good! Just a couple of things:

- For the input field labels - can you bold them?

- Can I ask about the name for the pod? I think it always defaults to "my-pod" even if you already have a "my-pod", so on the 2nd time you get an error. Is there a way to add a check and append a -01 / -02 / -03 ... / -$N on the end of the name?

lstocchi commented:
@mairin 

updated the PR. Now it looks like 
![image](https://user-images.githubusercontent.com/49404737/235377010-2f8475fd-b0b0-4d00-a8d8-daee713a14ac.png)

is it ok? Let me know, thanks!! :)

Regarding your second point, i'll open a new issue so we can work on it in a separate PR. This one is just for defining pod ports. I included a design refactoring bc i added a new element but i don't want to mix too many things.

mairin commented:
@lstocchi Looks great!!! Thanks for opening the other issue.
